src/rdkafka_fetcher.c

```diff
@@ -657,11 +807,24 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
 
                 rd_kafka_buf_read_arraycnt(rkbuf, &PartitionArrayCnt,
                                            RD_KAFKAP_PARTITIONS_MAX);
+                if (rd_kafka_buf_ApiVersion(request) >= 12) {
+                        rd_kafkap_Fetch_reply_tags_set_topic(
+                            &FetchTags, i, topic_id, PartitionArrayCnt);
+                }
 
                 for (j = 0; j < PartitionArrayCnt; j++) {
                         if (rd_kafka_fetch_reply_handle_partition(
-                                rkb, &topic, rkt, rkbuf, request, ErrorCode))
+                                rkb, &topic, rkt, rkbuf, request, ErrorCode,
+                                has_fetch_tags ? &FetchTags.Topics[i] : NULL,
+                                has_fetch_tags
+                                    ? &FetchTags.Topics[i].Partitions[j]
+                                    : NULL))
                                 goto err_parse;
+                        if (has_fetch_tags &&
+                            FetchTags.Topics[i]
+                                .partitions_with_leader_change_cnt) {
+                                FetchTags.topics_with_leader_change_cnt++;
+                        }
                 }
```

Comment:
This must be outside the for loop when all partitions have been read
```suggestion
                                goto err_parse;
                }
                if (has_fetch_tags &&
                    FetchTags.Topics[i].partitions_with_leader_change_cnt) {
                        FetchTags.topics_with_leader_change_cnt++;
                }
```

Reply:
Fixed